src/ext/java/no/entur/uttu/ext/fintraffic/security/FintrafficUserContextService.java

```diff
@@ -109,6 +114,65 @@ public class FintrafficUserContextService implements UserContextService {
    */
   private final ObjectMapper objectMapper;
 
+  private final LoadingCache<VacoApiCallContext, JsonNode> msGraphAppRoleAssignmentsCache =
+    CacheBuilder
+      .newBuilder()
+      .expireAfterAccess(20, TimeUnit.MINUTES)
+      .build(
+        new CacheLoader<>() {
+          @Override
+          public JsonNode load(VacoApiCallContext vacoApiCallContext) throws Exception {
+            MutableRequest request = MutableRequest
+              .GET(
+                MICROSOFT_GRAPH_API_ROOT +
+                "/users/" +
+                vacoApiCallContext.oid +
+                "/appRoleAssignments"
+              )
+              .header(
+                "Authorization",
+                "Bearer " + vacoApiCallContext.token.getAccessToken()
+              )
+              .header("Content-Type", "application/json");
+
+            HttpResponse<String> response = httpClient.send(
+              request,
+              BodyHandlers.ofString()
+            );
+            return objectMapper.readTree(response.body());
+          }
+        }
+      );
+
+  private final LoadingCache<VacoApiCallContext, VacoApiResponse<Me>> companyMembersCache =
+    CacheBuilder
+      .newBuilder()
+      .expireAfterAccess(20, TimeUnit.MINUTES)
+      .build(
+        new CacheLoader<>() {
+          @Override
+          public VacoApiResponse<Me> load(VacoApiCallContext vacoApiCallContext)
+            throws Exception {
```

Comment:
Maybe some logs here, if exception occurs or even without? (assuming the current weird situation with logs lost will get resolved)